Improvements of the filtering method

namsaraeva opened the pull request:
WIP

sophiamaedler committed:
add log message indicating how many GPUS are available for cellpose segementations

Commit:
update cell featurizer to calculate additional features

Commit:
add continue_segmentation method to SPARCSpy project to allow to continue segmenting failed sharded segmentations without recalculating already completed shards

Commit:
Merge branch 'development' of https://github.com/MannLabs/SPARCSpy into dev_sophia

sophiamaedler committed:
update cytosol_only method to cleanup memory, use memory mapped storage and optimze RAM usage

sophiamaedler committed:
initital commit adapting cytosol_only segmentation method to also utilize downsampling

Commit:
Removed refs to nuclei from cytosol only downsample cellpose seg

Commit:
write out shard location and check match before resolving sharding

Commit:
speedup of filtering

Commit:
Merge branch 'dev_sophia' of https://github.com/MannLabs/SPARCSpy into dev_sophia

Commit:
add timestamps

Commit:
add more messages

Commit:
add df

Commit:
add filtered datasets to sparcs-stat to get their dimensions displayed

Commit:
allow seeding of combine_datasets_balanced to generate reproducible dataloaders

Commit:
update transform for clearer language use and with more relevant sigma range

Commit:
update models to use properly labelled architecture, plus bug fixes

Commit:
add functionality to allow for classification of filtered datasets

Commit:
update extraction workflow to work with memory mapped temparrays and to utilize multiprocessing to improve speed performance

Commit:
revert changes to filtering workflow as speedup does not scale to larger datasets

sophiamaedler committed:
add log message saying where the input image is being looked for

sophiamaedler committed:
update logging for debugging

sophiamaedler committed:
further logging improvements

add further descriptive logging to try and isolate issue with 4 GPUs per node runs

sophiamaedler committed:
update time logging to include step numbers as seperate column in dataframe and designation of where logging csv should be saved to

sophiamaedler committed:
Merge branch 'dev_sophia' into timestamps_branch_altana

sophiamaedler commented:
I updated the logging format a bit to designate a clearer step id number and added this as a separate column to the data frame. In addition I merged my changes from `sophia_dev` to make sure we are working with the most recent version of the code. I think we would now be ready to do a test run and benchmark the datasets we discussed.

sophiamaedler committed:
add support for MPS device for cellpose segmentations running on M1 macs

note this still requires that pytorch and cellpose were installed with MPS backend enabled.

sophiamaedler committed:
check to see if labels dataset already exists when saving segmentation and if so remove.

sophiamaedler committed:
update .gitignore to not track development files created in repo

sophiamaedler committed:
move segmentation function call behind try and execpt statement to get better error log.

sophiamaedler committed:
add additional downsampling function to only select every nth pixel

sophiamaedler committed:
move code to save classes to seperate function for better readability, remove unnecessary code

sophiamaedler committed:
reformat cellpose segmentation to better understand filtering steps and make exection of filtering optional (is turned on by default)

in some cases it could be beneficial to simply out put both cytosol and nucleus mask from the cellpose segmentation method without applying the filtering directly to the mask. This is now possible as the filtering is implemented as an optional parameter.

sophiamaedler committed:
add framework to implement methods to filter a segmentatin mask to only include cells that pass certain filtering thresholds

this framework currently does not amend the segmentation masks but outputs a new classes file which contains nucleus_id:cytosol_id matching of all cells that pass filtering. This file is read in during extraction and used to update the information for which cells are extracted from the dataset.

sophiamaedler committed:
remove unecessary print statements

sophiamaedler committed:
add second example dataset

sophiamaedler committed:
update complete segmentation to fix errors in log messages and to allow for proper retiling

sophiamaedler committed:
add functionality to complete an incomplete segmentation filtering without needing to recalculate values for completed tiles
